featuretools/feature_base/feature_base.py

```diff
@@ -371,45 +371,47 @@ class DirectFeature(FeatureBase):
     input_types = [Variable]
     return_type = None
 
-    def __init__(self, base_feature, child_entity=None, relationship_path=None):
+    def __init__(self, base_feature, child_entity, relationship_path=None):
         """relationship_path is a forward path from child to parent."""
         base_feature = _check_feature(base_feature)
 
         self.parent_entity = base_feature.entity
 
+        relationship_path, self._path_is_unique = \
+            self._handle_relationship_path(child_entity, relationship_path)
+
+        super(DirectFeature, self).__init__(entity=child_entity,
+                                            base_features=[base_feature],
+                                            relationship_path=relationship_path,
+                                            primitive=PrimitiveBase)
+
+    def _handle_relationship_path(self, child_entity, relationship_path):
         if relationship_path:
             first_child = relationship_path[0].child_entity
-            if child_entity:
-                assert child_entity == first_child, \
-                    'child_entity must match the first relationship'
-            else:
-                child_entity = first_child
+            assert child_entity == first_child, \
+                'child_entity must match the first relationship'
 
             assert self.parent_entity == relationship_path[-1].parent_entity, \
                 'Base feature must be defined on the entity at the end of relationship_path'
 
-            self._is_unique_path = _is_unique_forward_path(child_entity.id,
-                                                           self.parent_entity.id,
-                                                           child_entity.entityset)
-        if not relationship_path:
-            assert child_entity, 'child_entity or relationship_path must be provided'
+            path_is_unique = child_entity.entityset \
```

Comment:
now that i look at this, it seem like `has_unique_forward_path` should be defined on Entity. 

So this should would be `child_entity.has_unique_forward_path(self.parent_entity.id)`

Reply:
you're right. let's leave as is